src/main/java/loanbook/logic/commands/RemindCommand.java

```diff
@@ -31,42 +29,39 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Send a reminder email to the customer.\n"
             + "Parameter: "
-            + PREFIX_PASSWORD + "EMAILPASSWORD "
-            + PREFIX_NAME + "NAME "
-            + PREFIX_BIKE + "BIKE\n"
+            + PREFIX_EMAILPW + "EMAILPASSWORD "
+            + PREFIX_ID + "ID "
             + "Example: " + COMMAND_WORD + " "
-            + PREFIX_PASSWORD + "samplepassword "
-            + PREFIX_NAME + "John Doe "
-            + PREFIX_BIKE + "BIKE001";
+            + PREFIX_EMAILPW + "samplepassword "
+            + PREFIX_ID + "0 ";
 
     public static final String MESSAGE_SUCCESS = "Email sent!";
 
-    private final Name name;
-    private final Bike bike;
+    private final LoanId id;
     private final String emailPassword;
 
     /**
      * Creates an RemindCommand to send a reminder email to customer's {@code Email}
      * according to the {@code Name} and {@code BikeId} provided.
      */
-    public RemindCommand(String emailPassword, Name name, Bike bike) {
-        requireAllNonNull(emailPassword, name, bike);
-        this.name = name;
-        this.bike = bike;
+    public RemindCommand(String emailPassword, LoanId id) {
+        requireAllNonNull(emailPassword, id);
+        this.id = id;
         this.emailPassword = emailPassword;
     }
 
     @Override
     public CommandResult execute(Model model, CommandHistory history) throws CommandException {
         requireNonNull(model);
-        List<Loan> lastShownList = model.getLoanBook().getLoanList();
+        List<Loan> loanList = model.getLoanBook().getLoanList();
+        int maxId = loanList.size() - 1;
 
-        Loan targetLoan = getLoan(lastShownList, name, bike);
-
-        if (targetLoan == null) {
-            throw new CommandException(String.format(Messages.MESSAGE_INVALID_INFO, name, bike.getName()));
+        if (id.value > maxId) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INFO);
         }
 
+        Loan targetLoan = loanList.get(id.value);
```

Comment:
This is actually very dependent on the fact that the Loans in the loanList start from ID = 0 and are in ascending order in the list without gaps. Gaps can form when the user deletes Loans, since the current implementation literally removes it from the list of Loans.

A better approach would be have a method in the Model, possibly called `getLoanById(LoanId)`.

I will be implementing this and getting it out in a PR by tonight.

Reply:
I have created and merged the relevant functions you need.

Let me know if you need anything else.

Reply:
Thank you so much!